pkg/generate/app/pipeline.go

```diff
@@ -39,20 +39,32 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 
 // NewBuildPipeline creates a new pipeline with components that are expected to
 // be built.
-func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
-	name, ok := NameSuggestions{source, input}.SuggestName()
-	if !ok {
-		return nil, ErrNameRequired
-	}
+func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef, to string) (*Pipeline, error) {
+	var outputImageRef image.DockerImageReference
 
-	output := &ImageRef{
-		Reference: image.DockerImageReference{
+	if len(to) > 0 {
+		var err error
+		outputImageRef, err = image.ParseDockerImageReference(to)
```

Comment:
The pipeline output has a [`Reference`](https://github.com/openshift/origin/blob/aa8e6ba1e302ff7f5d89425d92b1e73c4979df0d/pkg/generate/app/imageref.go#L134) field of type `imageapi.DockerImageReference`, which [later on gets converted](https://github.com/openshift/origin/blob/aa8e6ba1e302ff7f5d89425d92b1e73c4979df0d/pkg/generate/app/imageref.go#L210-L229) to a `kapi.ObjectReference`, of `Kind` being either `ImageStreamTag` or `DockerImage`.

I believe we use a `imageapi.DockerImageReference` because it can, for what is needed here, represent both ImageStreamTag and DockerImage references, given that the format of ISTs is a subset of that of a DockerImage.